x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/asset_criticality_data_client.ts

```diff
@@ -291,11 +310,18 @@ export class AssetCriticalityDataClient {
       return undefined;
     }
 
+    if (record.id_value === CRITICALITY_VALUES.DELETED) {
```

Comment:
Do you think this check is necessary? The `if` before this one already checks if the record is undefined, and from what I understand, `this.get` should return undefined for deleted records.

Reply:
🤔 I kinda like the fact that the `delete` method includes an explicit check for the `deleted` case. OTOH, I hate that it's "dead code", so I will remove it I think